parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -200,103 +201,143 @@ downloadSecureTLSBootstrapKubeletExecPlugin() {
-                local shim_filename="${shim_prefix}${shim}${version_suffix}"
-                retrycmd_if_failure 2 5 60 curl -fSLv -o "${containerd_wasm_filepath}/${shim_prefix}${shim}-${binary_version}${version_suffix}" "${base_url}/${shim_filename}" 2>&1 | tee "$CURL_OUTPUT" >/dev/null | grep -E "^(curl:.*)|([eE]rr.*)$" && (cat "$CURL_OUTPUT" && exit $ERR_KRUSTLET_DOWNLOAD_TIMEOUT) &
-                wasmShimPids+=($!)
-            done
-        fi
+    package_versions=()
+    for version in "${json_versions[@]}"; do
+        package_versions+=("v$version")
     done
 
-    wait ${wasmShimPids[@]}
-
-    # set permissions for the shims
-    for shim_version in $CONTAINERD_WASM_VERSIONS; do
-        binary_version="$(echo "${shim_version}" | tr . -)"
-        if [ "$shim_version" == "v0.15.1" ]; then
-            chmod 755 "$containerd_wasm_filepath/containerd-shim-spin-${binary_version}-v2"
-            # spin shim v0.15.1 cannot be renamed: https://github.com/spinkube/containerd-shim-spin/issues/190
-            # so we rename the shim back to containerd-shim-spin-v2
-            mv "$containerd_wasm_filepath/containerd-shim-spin-${binary_version}-v2" "$containerd_wasm_filepath/containerd-shim-spin-v2"
-        else    
-            chmod 755 "$containerd_wasm_filepath/containerd-shim-spin-${binary_version}-v1"
-            chmod 755 "$containerd_wasm_filepath/containerd-shim-slight-${binary_version}-v1"
-            if [ "$shim_version" == "v0.8.0" ]; then
-                chmod 755 "$containerd_wasm_filepath/containerd-shim-wws-${binary_version}-v1"
-            fi
-        fi
+    for version in "${package_versions[@]}"; do
+        containerd_wasm_url=$(evalPackageDownloadURL ${PACKAGE_DOWNLOAD_URL})
+        downloadSpinKube $download_location $containerd_wasm_url $version
     done
+    wait ${SPINKUBEPIDS[@]}
+    for version in "${package_versions[@]}"; do
+        updateSpinKubePermissions $download_location $version
+    done
+}
+
+downloadSpinKube(){
+    local containerd_wasm_filepath=${1}
+    local containerd_wasm_url=${2}
+    local shim_version=${3}
+    local binary_version="$(echo "${shim_version}" | tr . -)"
+
+    if wasmFilesExist "$containerd_wasm_filepath" "$shim_version" "spin" "-v2"; then
+        return
+    fi
+
+    BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER:=}"
+    if [[ ! -z ${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER} ]]; then
+        local registry_url="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER}/oss/binaries/spinkube/containerd-shim-spin:${shim_version}-linux-${CPU_ARCH}"
+        local wasm_shims_tgz_tmp="${containerd_wasm_filepath}/containerd-shim-spin-v2"
+        retrycmd_get_binary_from_registry_with_oras 120 5 "${wasm_shims_tgz_tmp}" "${registry_url}" || exit $ERR_ORAS_PULL_CONTAINERD_WASM
+        mv "${containerd_wasm_filepath}/containerd-shim-spin-v2" "${containerd_wasm_filepath}/containerd-shim-spin-${binary_version}-v2"
```

Comment:
suggest: This line is not needed anymore, we also need to update `updateSpinKubePermissions` to remove the re-naming line, and change the output path from`retrycmd_if_failure 30 5 60 curl -fSLv -o "$containerd_wasm_filepath/containerd-shim-spin-${binary_version}-v2" "$containerd_wasm_url/containerd-shim-spin-v2"` to  `$containerd_wasm_filepath/containerd-shim-spin-v2`.